pkg/providers/instancetype/skus.go

```diff
@@ -0,0 +1,3306 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instancetype
+
+import (
+	"github.com/Azure/skewer"
+	"k8s.io/utils/ptr"
+)
+
+// getKarpenterWorkingSKUs returns a the list of SKUs that are
+// allowed to be used by Karpenter. This is a subset of the
+// SKUs that are available in Azure.
+func getKarpenterWorkingSKUs() []skewer.SKU {
+	workingSKUs := []skewer.SKU{}
+	for _, sku := range allAzureVMSkus {
+		var found bool
+		// If we find this SKU in the AKS restricted list, exclude it
+		for _, aksRestrictedSKU := range aksRestrictedVMSKUs {
+			if aksRestrictedSKU.Name != nil && sku.Name != nil {
+				if *sku.Name == *aksRestrictedSKU.Name {
+					found = true
+				}
+			}
+		}
+		// If it's not in the AKS restricted list, it may be in the Karpenter restricted list
+		if !found {
+			for _, karpenterRestrictedSKU := range karpenterRestrictedVMSKUs {
+				if karpenterRestrictedSKU.Name != nil && sku.Name != nil {
+					if *sku.Name == *karpenterRestrictedSKU.Name {
+						found = true
+					}
+				}
+			}
+		}
+		// If it's not in any of the restricted lists, we register it as a working VM SKU
+		if !found {
+			workingSKUs = append(workingSKUs, sku)
+		}
+	}
+	return workingSKUs
+}
+
+// aksRestrictedVMSKUs is a list of VM SKUs that are not allowed by AKS.
+var aksRestrictedVMSKUs = []skewer.SKU{
```

Comment:
We should already have this list in https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/providers/instancetype/restricted.go#L22

Reply:
ACK, re-used the existing var, thanks! (although I did move it to the new `pkg/providers/instancetype/skus.go` file)